tony-history-server/app/utils/ParserUtils.java

```diff
@@ -0,0 +1,88 @@
+package utils;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.gson.JsonSyntaxException;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import models.JobConfig;
+import models.JobMetadata;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import play.Logger;
+import play.libs.Json;
+
+import static utils.HdfsUtils.*;
+
+/**
+ * The class handles parsing different file format
+ */
+public class ParserUtils {
+  private static final Logger.ALogger LOG = Logger.of(ParserUtils.class);
+
+  public static JobMetadata parseMetadata(FileSystem fs, Path metricsFilePath) {
+    if (!isPathValid(fs, metricsFilePath)) {
+      return new JobMetadata();
+    }
+
+    String fileContent = contentOfHdfsFile(fs, metricsFilePath);
+    JobMetadata jobMetadata = new JobMetadata();
+    JsonNode jObj;
+    try {
+      jObj = Json.parse(fileContent);
+    } catch (JsonSyntaxException e) {
+      LOG.error("Couldn't parse metadata", e);
+      return jobMetadata;
+    }
+    jobMetadata.setId(jObj.get("id").textValue());
+    jobMetadata.setJobLink(jObj.get("url").textValue());
+    jobMetadata.setConfigLink("/jobs/" + jobMetadata.getId());
+    jobMetadata.setStarted(jObj.get("started").textValue());
+    jobMetadata.setCompleted(jObj.get("completed").textValue());
+    LOG.info("Successfully parsed metadata");
+    return jobMetadata;
+  }
+
+  public static List<JobConfig> parseConfig(FileSystem fs, Path configFilePath) {
+    if (!isPathValid(fs, configFilePath)) {
+      return new ArrayList<>();
+    }
+
+    File configFile = new File(copyFromHdfs(fs, configFilePath));
```

Comment:
You shouldn't need to copy the file from HDFS to local. DocumentBuilder has a .parse() method that takes an InputStream. You should be able to open the HDFS file and feed the InputStream into DocumentBuilder.

Reply:
Fixed: https://github.com/linkedin/TonY/pull/62/commits/3c82d8a3ef4c31f9d8c671e734d4b1853fbb1b2a